### DISCUSSION TITLE
REM

### DISCUSSION THREAD

**Discussion by davthomaspilot** [tagyoureit/nodejs-poolController#940] [REM]
I posted on REM github "issues", but I think I should have posted here.  Any help would be appreciated! 

[davthomaspilot](https://github.com/davthomaspilot) commented [6 hours ago](https://github.com/rstrouse/relayEquipmentManager/issues/60#issue-2273831837)
image

I've had a problem for a couple of weeks with PH not dosing. pH pump would turn on/off as expected, but ph wouldn't change. So, I've been dosing acid manually.

I replaced the pump tube and one way injector valve, and still it seemed no acid was making it in the pool (and actual acid tank level wasn't changing).

Today, I was going to disconnect the tube at the check valve to see if the acid actually made it that far. But, I could not find a way to manually dose. Then I noticed a "no flow" message.

Flow is actually good, so I was heading to get a DVM to see if the sensor was actually indicating flow and rebooted both the REM pi and the njsPC pi. When it came back up, I saw probe and pump faults for both ph and ORP.

What could cause this?

Note the ph and ORP readings are still updating and look believable.

Also, the tank levels I just set didn't seem to survive the reboot.

@[davthomaspilot](https://github.com/davthomaspilot)
Author
davthomaspilot commented [5 minutes ago](https://github.com/rstrouse/relayEquipmentManager/issues/60#issuecomment-2089289513)
I used pigpio (pigs) to check state of the flow switch GPIO. It changes as expected when I turn the pool pump on and off.

Are there any diagnostics I can do in the web interface to REM? For some reason, the state of that GPIO is seen by the rpi (pigpio) but not REM.

Probably same root cause is causing the faults.

**Comment by rstrouse** [comment]
Open the chemistry configuration in dashPanel and make sure the devices are selected on the hardware page.

**Comment by davthomaspilot** [comment]
Wow, they were not! 

And when I try to select the ph Probe (click the checkbox), it clears when I do the save.

I guess that means for some reason, the configuration got lost

Also, the Flow Sensor field was empty, so I set it to be "Flow Switch"). 

Before I click Save:

![image](https://github.com/tagyoureit/nodejs-poolController/assets/6830886/0be2f748-7302-425b-a22f-0beff34eadaf)


After: 

![image](https://github.com/tagyoureit/nodejs-poolController/assets/6830886/ca280b42-aa02-46a9-9ea0-40f0c346b814)

**Comment by davthomaspilot** [comment]
That cleared the flow sensor fault, but I still have the pH Pump Fault and ORP Pump Fault. 

Probably some other configuration has been lost, I'll go down that path (it's been a while since I messed with this stuff).

**Comment by davthomaspilot** [comment]
When I revisited the Chemistry-Hardware page, the REM Feed box is now checked.  

Also, the pump faults are likely due to the  Device settings for the ORP Pump and Ph Pumb being lost too.  So, I set those up again, and save d the controller.


That fixed everything.  Thanks!

So, does that happen sporadically (lose the hardware settings)?

**Comment by davthomaspilot** [comment]
Everything was working, but I'm seeing the pump fault messages again.

And, the configuration is again lost.

What would cause this, should I re-install?

Thanks

**Reply by tagyoureit** [reply]
First, make sure you’re SD card hasn’t run out of space.
